docs/languages/csharp/expression-trees-execution.md

```diff
@@ -0,0 +1,194 @@
+that tries to determine of two different expression trees result in
+the same executable code.
+
+## Caveats
+
+Compiling a lambda expression to a delegate and invoking that delegate
+is one of the simplest operations you can perform with an expression
+tree. However, even with this simple operation, there are caveats
+you must be aware of. 
+
+Lambda Expressions create closures aver any local variables that are
+referenced in the expression. You must guarantee that any variables
+that would be part of the delegate are usable at the location where
+you call `Compile`, and when you execute the resulting delegate.
+
+In general, the compiler will ensure that this is true. However,
+if your expression accesses a variable that implememts `IDisposable`,
+it's possible that your code might dispose of the object while it
+is still held by the expression tree.
+
+For example, this code works fine, because `int` does not implement
+`IDisposable`:
+
+```cs
+private static Func<int, int> CreateBoundFunc()
+{
+    var constant = 5; // constant is captured by the expression tree
+    Expression<Func<int, int>> expression = (b) => constant + b;
+    var rVal = expression.Compile();
+    return rVal;
+}
+```
+
+The delegate has captured a reference to the local variable `constant`.
+That variable is accessed at any time later, when the function returned
+by `CreateBoundFunc` executes.
+
+However, consider this (rather contrived) class that implements
+`IDisposable`:
+
+```cs
+public class Resource : IDisposable
+{
+    private bool isDisposed = false;
+    public int Argument
+    {
+        get
+        {
+            if (!isDisposed)
+                return 5;
+            else throw new ObjectDisposedException("Resource");
+        }
+    }
+
+    public void Dispose()
+    {
+        isDisposed = true;
+    }
+}
+```
+
+If you use it in an expression as shown below, you'll get an
+`ObjectDisposedException` when you execute the code referenced
+by the `Resource.Argument` property:
+
+```cs
+private static Func<int, int> CreateBoundResource()
+{
+    using (var constant = new Resource()) // constant is captured by the expression tree
+    {
+        Expression<Func<int, int>> expression = (b) => constant.Argument + b;
+        var rVal = expression.Compile();
+        return rVal;
+    }
+}
+```
+
+The delegate returned from this method has closed over the `constant` object,
+which has been disposed of. (It's been disposed, because it was declared in a
+`using` statement.) 
+
+Now, when you execute the delegate returned from this method, you'll have a
+`ObjecctDisposedException` thrown at the point of execution.
+
+It does seem strange to have a runtime error representing a compile-time
+construct, but that's the world we enter when we work with
+expression trees.
+
+There are a lot of permutations of this problem, so it's hard
+to offer general guidance to avoid it. The best that I can say is
```

Comment:
So far I've been writing docs without using the word "I".  It feels a little jarring to see it when 99% of the docs don't speak from the first person.  Might be worth phrasing this a little differently.